src/main.rs

```diff
@@ -63,7 +73,17 @@ fn main() -> Result<()> {
                 }
             };
             let env = matches.value_of("env").map(|v| v.to_owned());
-            println!("{}", reqq.execute(req, env)?);
+            let req_args = matches.values_of("arg").unwrap();
+            let mut extra_args: HashMap<&str, &str> = HashMap::new();
+            for arg in req_args {
+                let kv_pair: Vec<&str> = arg.splitn(2, "=").collect();
+                // I'm sure there is a better Rust way to do this.
+                // If the arg key/value pair is malformed this should
+                // gracefully fail. Not sure how to best to protect against
+                // the failure.
```

Comment:
Yeah this is tricky! What constitutes malformed? If the k=v pair is quoted, it could have string and symbols in it that should be valid strings embeddable into the template. Really it comes down to whether the template is able to be executed successfully.

Thinking back on your other comment about making the value more generic -- handlebars is a powerful templating language so having access to all JSON types would be cool, but things are simplified on the parsing and executing level if we assume all input (for extra args at least) are always strings. What do you think?

Reply:
Yeah, I agree on the extra args being strings. I think strings should get the result needed in most cases unless you want to be able to do things like `{{ arg1 + arg2 }}` in the template.

For an arg to be malformed I think the '=' would not exist in the string.